src/components/UserAvatar.js

```diff
@@ -1,40 +1,45 @@
-import { Box, Link, Avatar, Typography } from '@mui/material';
+import { Box, Link, Avatar, Typography, Tooltip } from '@mui/material';
 
 function UserAvatar({ auth }) {
   if (!auth?.isAuthenticated) {
     return null;
   }
+
+  const userName = auth?.user?.profile.preferred_username[0] || null;
+
   return (
-    <Box>
-      <Link
-        href="https://dev-k8s.treetracker.org/keycloak/realms/treetracker/account/#/personal-info"
-        target="_blank"
-        rel="noopener noreferrer"
-        sx={{
-          color: ({ palette }) => palette.text.primary,
-        }}
-      >
-        <Avatar
+    <Tooltip title={userName} arrow>
+      <Box>
+        <Link
+          href="https://dev-k8s.treetracker.org/keycloak/realms/treetracker/account/#/personal-info"
+          target="_blank"
+          rel="noopener noreferrer"
           sx={{
-            width: 24,
-            height: 24,
-            background: ({ palette }) => palette.background.greenGradient,
+            color: ({ palette }) => palette.text.primary,
           }}
         >
-          <Typography
-            variant="caption"
+          <Avatar
             sx={{
-              textTransform: 'uppercase',
-              fontSize: 16,
-              textDecoration: 'none',
-              color: ({ palette }) => palette.text.contrast,
+              width: 24,
+              height: 24,
+              background: ({ palette }) => palette.background.greenGradient,
             }}
           >
-            {auth?.user?.profile.preferred_username[0] || null}
-          </Typography>
-        </Avatar>
-      </Link>
-    </Box>
+            <Typography
+              variant="caption"
+              sx={{
+                textTransform: 'uppercase',
+                fontSize: 16,
+                textDecoration: 'none',
+                color: ({ palette }) => palette.text.contrast,
+              }}
+            >
+              {userName}
```

Comment:
We can change this to {userName ? userName[0] : null} to conform with the idea on line 8.